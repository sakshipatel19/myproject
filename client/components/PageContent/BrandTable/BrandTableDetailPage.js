import React, { Component, Fragment } from "react";
import "./BrandTable.scss";
import LinearProgressBar from "../../common/LinearProgressBar";
import DotWithLabel from "../../common/DotWithLabel";
import Icon from "../../common/Icon";
import { applySort } from "../../../utils/array";

const SORT_ORDER = {
	ascendingOrder: "ASC",
	descendingOrder: "DESC"
};

class BrandTableDetailPage extends Component {
	state = {
		showBestSellerBrandDetailPage: true,
		sortType: "ASC",
		sortBy: "label"
	};

	createTableHeaderObject = () => {
		return [
			{
				name: "Name",
				key: "label"
			},
			{
				name: "Score",
				key: "score",
				isNumber: true
			}
		];
	};

	bindHeaderRowData = (arrObj, dataLength) => {
		const headersList = arrObj.map((e, i) => {
			return (
				<div
					className={`${`brand-type-col-${i + 1}`} ${this.state.sortBy === e.key && dataLength > 5 ? "sortSelected" : ""
						} ${e.isNumber ? "number-text" : ""} ${dataLength <= 5 ? "disableSort" : ""
						}`}
					onClick={() => this.onSortChange(e.key)}
					key={`${e.key}`}
				>
					{e.name}
					<span className="sort-icon">
						<Icon
							iconClass={`${this.state.sortBy === e.key &&
								dataLength > 5 &&
								"visible"}`}
							name={this.state.sortType === "ASC" ? "sortUp" : "sortDown"}
							size={8}
						/>
					</span>
				</div>
			);
		});

		return <div className="brand-type-row header">{headersList}</div>;
	};

	onSortChange = sortBy => {
		let currentsortOrder = this.state.sortType;
		let sortType = SORT_ORDER.ascendingOrder;
		let currentSort = this.state.sortBy;
		if (currentSort === sortBy)
			sortType =
				currentsortOrder === SORT_ORDER.ascendingOrder
					? SORT_ORDER.descendingOrder
					: SORT_ORDER.ascendingOrder;
		this.setState({ sortType, sortBy });
	};

	createBrandTableBody = data => {
		const brandTableList = data?.brandScoreInitial || [];
		const brandList = brandTableList.map((item, i) => {
			return (
				<div className="brand-type-row" key={`${item?.label}_${i}`}>
					<div className={`brand-type-col-1`}>{item.label}</div>
					{item.compareScore !== null ? (
						<div className="brand-type-col-3 compare-linear-progressbar">
							<LinearProgressBar
								rounded
								animate
								percent={item.score / 100}
								width={674}
								height={5}
								backgroundColor="#f3f5f8"
								fillColor={"#4CA0FA"}
							/>
							<LinearProgressBar
								rounded
								animate
								percent={item.compareScore / 100}
								width={674}
								height={4.8}
								backgroundColor="#f3f5f8"
								fillColor="#F79764"
							/>
						</div>
					) : (
							<div className="brand-type-col-3">
								<LinearProgressBar
									rounded
									animate
									percent={item.score / 100}
									width={674}
									height={5}
									backgroundColor="#f3f5f8"
									fillColor={"#4CA0FA"}
								/>
							</div>
						)}
					<div className={`brand-type-col-2`}>
						{this.props.compareSelected ? (
							<div className="dots-container">
								<Fragment>
									<DotWithLabel
										color="#4CA0FA"
										text={item?.score === null ? "NA" : `${item?.score}%`}
									/>
									<DotWithLabel
										color="#F79764"
										text={
											item?.compareScore === null
												? "NA"
												: `${item?.compareScore}%`
										}
									/>
								</Fragment>
							</div>
						) : (
								<div>
									{item?.score === null || item?.score === undefined
										? "NA"
										: `${item?.score}%`}
								</div>
							)}
					</div>
				</div>
			);
		});
		return <div className="brand-type-row-content">{brandList}</div>;
	};

	render() {
		let data = this.props.data || [];
		let brandScoreInitial = data?.brandScoreInitial || [];
		applySort(
			brandScoreInitial,
			this.state.sortBy,
			this.state.sortType,
			SORT_ORDER
		);
		return (
			<div className="detailpage-container">
				{this.state.showBestSellerBrandDetailPage && (
					<div className="detailpage-table-body">
						{this.bindHeaderRowData(
							this.createTableHeaderObject(),
							brandScoreInitial?.length
						)}
						{this.createBrandTableBody(data)}
					</div>
				)}
			</div>
		);
	}
}

export default BrandTableDetailPage;
