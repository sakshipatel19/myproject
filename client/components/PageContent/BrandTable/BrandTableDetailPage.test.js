import React from "react";
import { shallow } from "enzyme";

import BrandTableDetailPage from "./BrandTableDetailPage";

const data = {
    heading:"Brands",
    scorehelp:"Content scores for your selected brands",
    brandScoreInitial:[
        {
            label:"Advil",
            score:68,
            compareScore:null
        },
        {
            label:"alli",
            score:70,
            compareScore:70
        },
        {
            label:"Centrum",
            score:null,
            compareScore:60
        },
        {
            label:"Aquafresh",
            score:null,
            compareScore:null
        }
    ]
};
const fetchBrands = {
    type:"GET_CONTENT_BRANDS_SCORE_REQUEST",
    value:{}
}
let baseProps = {
    data:data,
    fetchBrands:jest.fn(),
    compareSelected:false,
};
describe("BrandTableDetailPage tests", () => {
    it("BrandTableDetailPage should render correctly", () => {
        const wrapper = shallow(<BrandTableDetailPage {...baseProps} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("BrandTableDetailPage should render correctly when data received is null", () => {
        const wrapper = shallow(<BrandTableDetailPage data={null} fetchBrands={fetchBrands}/>);
        expect(wrapper).toMatchSnapshot();
    });
    it("button click", () => {
        const wrapper = shallow(<BrandTableDetailPage {...baseProps}/>);
        wrapper.find('.brand-type-col-1.sortSelected').simulate('click');
        expect(wrapper.state('sortType')).toBe("DESC");
        wrapper.find('.brand-type-col-1.sortSelected').simulate('click');
        expect(wrapper.state('sortType')).toBe("ASC");
        
        //to check negative scenario for condition (currentSort === sortBy) in onSortChange function
        wrapper.setState({sortBy:'score'});
        wrapper.find('.brand-type-row.header>.brand-type-col-1').simulate('click');
        expect(wrapper.state('sortType')).toBe("ASC");
    });
    it("BrandTableDetailPage should render correctly when compare date is applied", () => {
        baseProps = {
            data:data,
            fetchBrands:jest.fn(),
            compareSelected:true,
        };
        const wrapper = shallow(<BrandTableDetailPage {...baseProps} />);
        expect(wrapper).toMatchSnapshot();
    });
});
