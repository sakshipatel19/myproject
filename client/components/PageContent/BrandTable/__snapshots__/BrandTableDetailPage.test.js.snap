// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandTableDetailPage tests BrandTableDetailPage should render correctly 1`] = `
<div
  className="detailpage-container"
>
  <div
    className="detailpage-table-body"
  >
    <div
      className="brand-type-row header"
    >
      <div
        className="brand-type-col-1 sortSelected "
        key="label"
        onClick={[Function]}
      >
        Name
        <span
          className="sort-icon"
        >
          <Icon
            iconClass="table-sort-image"
            name="sortUp"
            size={18}
          />
        </span>
      </div>
      <div
        className="brand-type-col-2  number-text"
        key="score"
        onClick={[Function]}
      >
        Score
        <span
          className="sort-icon"
        />
      </div>
    </div>
    <div
      className="brand-type-row-content"
    >
      <div
        className="brand-type-row"
        key="Advil_0"
      >
        <div
          className="brand-type-col-1"
        >
          Advil
        </div>
        <div
          className="brand-type-col-3"
        >
          <LinearProgressBar
            animate={true}
            backgroundColor="#f3f5f8"
            fillColor="#4CA0FA"
            height={5}
            percent={0.68}
            rounded={true}
            width={500}
          />
        </div>
        <div
          className="brand-type-col-2"
        >
          <div>
            68%
          </div>
        </div>
      </div>
      <div
        className="brand-type-row"
        key="alli_1"
      >
        <div
          className="brand-type-col-1"
        >
          alli
        </div>
        <div
          className="brand-type-col-3 compare-linear-progressbar"
        >
          <LinearProgressBar
            animate={true}
            backgroundColor="#f3f5f8"
            fillColor="#4CA0FA"
            height={5}
            percent={0.7}
            rounded={true}
            width={500}
          />
          <div
            className="compare-linear-progressbar"
          >
            <LinearProgressBar
              animate={true}
              fillColor="#F79764"
              height={5}
              percent={0.7}
              rounded={true}
              width={500}
            />
          </div>
        </div>
        <div
          className="brand-type-col-2"
        >
          <div>
            70%
          </div>
        </div>
      </div>
      <div
        className="brand-type-row"
        key="Aquafresh_2"
      >
        <div
          className="brand-type-col-1"
        >
          Aquafresh
        </div>
        <div
          className="brand-type-col-3"
        >
          <LinearProgressBar
            animate={true}
            backgroundColor="#f3f5f8"
            fillColor="#4CA0FA"
            height={5}
            percent={0}
            rounded={true}
            width={500}
          />
        </div>
        <div
          className="brand-type-col-2"
        >
          <div>
            NA
          </div>
        </div>
      </div>
      <div
        className="brand-type-row"
        key="Centrum_3"
      >
        <div
          className="brand-type-col-1"
        >
          Centrum
        </div>
        <div
          className="brand-type-col-3 compare-linear-progressbar"
        >
          <LinearProgressBar
            animate={true}
            backgroundColor="#f3f5f8"
            fillColor="#4CA0FA"
            height={5}
            percent={0}
            rounded={true}
            width={500}
          />
          <div
            className="compare-linear-progressbar"
          >
            <LinearProgressBar
              animate={true}
              fillColor="#F79764"
              height={5}
              percent={0.6}
              rounded={true}
              width={500}
            />
          </div>
        </div>
        <div
          className="brand-type-col-2"
        >
          <div>
            NA
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BrandTableDetailPage tests BrandTableDetailPage should render correctly when compare date is applied 1`] = `
<div
  className="detailpage-container"
>
  <div
    className="detailpage-table-body"
  >
    <div
      className="brand-type-row header"
    >
      <div
        className="brand-type-col-1 sortSelected "
        key="label"
        onClick={[Function]}
      >
        Name
        <span
          className="sort-icon"
        >
          <Icon
            iconClass="table-sort-image"
            name="sortUp"
            size={18}
          />
        </span>
      </div>
      <div
        className="brand-type-col-2  number-text"
        key="score"
        onClick={[Function]}
      >
        Score
        <span
          className="sort-icon"
        />
      </div>
    </div>
    <div
      className="brand-type-row-content"
    >
      <div
        className="brand-type-row"
        key="Advil_0"
      >
        <div
          className="brand-type-col-1"
        >
          Advil
        </div>
        <div
          className="brand-type-col-3"
        >
          <LinearProgressBar
            animate={true}
            backgroundColor="#f3f5f8"
            fillColor="#4CA0FA"
            height={5}
            percent={0.68}
            rounded={true}
            width={500}
          />
        </div>
        <div
          className="brand-type-col-2"
        >
          <div
            className="dots-container"
          >
            <DotWithLabel
              color="#4CA0FA"
              text="68%"
            />
            <DotWithLabel
              color="#F79764"
              text="NA"
            />
          </div>
        </div>
      </div>
      <div
        className="brand-type-row"
        key="alli_1"
      >
        <div
          className="brand-type-col-1"
        >
          alli
        </div>
        <div
          className="brand-type-col-3 compare-linear-progressbar"
        >
          <LinearProgressBar
            animate={true}
            backgroundColor="#f3f5f8"
            fillColor="#4CA0FA"
            height={5}
            percent={0.7}
            rounded={true}
            width={500}
          />
          <div
            className="compare-linear-progressbar"
          >
            <LinearProgressBar
              animate={true}
              fillColor="#F79764"
              height={5}
              percent={0.7}
              rounded={true}
              width={500}
            />
          </div>
        </div>
        <div
          className="brand-type-col-2"
        >
          <div
            className="dots-container"
          >
            <DotWithLabel
              color="#4CA0FA"
              text="70%"
            />
            <DotWithLabel
              color="#F79764"
              text="70%"
            />
          </div>
        </div>
      </div>
      <div
        className="brand-type-row"
        key="Aquafresh_2"
      >
        <div
          className="brand-type-col-1"
        >
          Aquafresh
        </div>
        <div
          className="brand-type-col-3"
        >
          <LinearProgressBar
            animate={true}
            backgroundColor="#f3f5f8"
            fillColor="#4CA0FA"
            height={5}
            percent={0}
            rounded={true}
            width={500}
          />
        </div>
        <div
          className="brand-type-col-2"
        >
          <div
            className="dots-container"
          >
            <DotWithLabel
              color="#4CA0FA"
              text="NA"
            />
            <DotWithLabel
              color="#F79764"
              text="NA"
            />
          </div>
        </div>
      </div>
      <div
        className="brand-type-row"
        key="Centrum_3"
      >
        <div
          className="brand-type-col-1"
        >
          Centrum
        </div>
        <div
          className="brand-type-col-3 compare-linear-progressbar"
        >
          <LinearProgressBar
            animate={true}
            backgroundColor="#f3f5f8"
            fillColor="#4CA0FA"
            height={5}
            percent={0}
            rounded={true}
            width={500}
          />
          <div
            className="compare-linear-progressbar"
          >
            <LinearProgressBar
              animate={true}
              fillColor="#F79764"
              height={5}
              percent={0.6}
              rounded={true}
              width={500}
            />
          </div>
        </div>
        <div
          className="brand-type-col-2"
        >
          <div
            className="dots-container"
          >
            <DotWithLabel
              color="#4CA0FA"
              text="NA"
            />
            <DotWithLabel
              color="#F79764"
              text="60%"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BrandTableDetailPage tests BrandTableDetailPage should render correctly when data received is null 1`] = `
<div
  className="detailpage-container"
>
  <div
    className="detailpage-table-body"
  >
    <div
      className="brand-type-row header"
    >
      <div
        className="brand-type-col-1 sortSelected "
        key="label"
        onClick={[Function]}
      >
        Name
        <span
          className="sort-icon"
        >
          <Icon
            iconClass="table-sort-image"
            name="sortUp"
            size={18}
          />
        </span>
      </div>
      <div
        className="brand-type-col-2  number-text"
        key="score"
        onClick={[Function]}
      >
        Score
        <span
          className="sort-icon"
        />
      </div>
    </div>
    <div
      className="brand-type-row-content"
    />
  </div>
</div>
`;
